homeassistant/components/crownstone/helpers.py

```diff
@@ -0,0 +1,52 @@
+"""Helper functions for the Crownstone integration."""
+from __future__ import annotations
+
+import os
+
+from serial.tools.list_ports_common import ListPortInfo
+
+from .const import DONT_USE_USB, MANUAL_PATH, REFRESH_LIST
+
+
+def list_ports_as_str(
+    serial_ports: list[ListPortInfo], no_usb_option: bool = True
+) -> list[str]:
+    """
+    Represent currently available serial ports as string.
+
+    Adds option to not use usb on top of the list,
+    option to use manual path or refresh list at the end.
+    """
+    ports_as_string: list[str] = []
+
+    if no_usb_option:
+        ports_as_string.append(DONT_USE_USB)
+
+    for port in serial_ports:
+        ports_as_string.append(
```

Comment:
Can you use `usb.human_readable_device_name`?

Reply:
It does appear the `vid` and `pid` fields of that function are not converted to hex however. They now show up as integers as described [here](https://pyserial.readthedocs.io/en/latest/tools.html#serial.tools.list_ports.ListPortInfo.vid) is that intentional? basically all usb listing tools show the hex representation.  Also the description string shows the product name twice, for example: `Crownstone dongle - Crownstone dongle`, while the `product` field actually shows this once.

Reply:
Does the device actually have the name in the description twice or is that a bug in the function? If it’s a bug in the function, we can accept a pr to fix it 

The function is expected the vid and pid to be passed in as a hex string.  Conversion example in:
homeassistant/components/usb/utils.py

Reply:
```console
Python 3.9.7 (default, Aug 31 2021, 13:28:12)
[GCC 11.1.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import serial.tools.list_ports
>>> ports = [(port.device, port.description, port.product) for port in serial.tools.list_ports.comports()]
>>> ports
[('/dev/ttyUSB0', 'Crownstone dongle - Crownstone dongle', 'Crownstone dongle')]
```

as well as
```console
$ lsusb -v -d 10C4:EA60

Bus 003 Device 006: ID 10c4:ea60 Silicon Labs CP210x UART Bridge
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass            0
  bDeviceSubClass         0
  bDeviceProtocol         0
  bMaxPacketSize0        64
  idVendor           0x10c4 Silicon Labs
  idProduct          0xea60 CP210x UART Bridge
  bcdDevice            1.00
  iManufacturer           1 Silicon Labs
  iProduct                2 Crownstone dongle
  iSerial                 3 01678887
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x0020
    bNumInterfaces          1
    bConfigurationValue     1
    iConfiguration          0
    bmAttributes         0x80
      (Bus Powered)
    MaxPower              100mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           2
      bInterfaceClass       255 Vendor Specific Class
      bInterfaceSubClass      0
      bInterfaceProtocol      0
      iInterface              2 Crownstone dongle
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x01  EP 1 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x82  EP 2 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               0
Device Status:     0x0000
  (Bus Powered)
```

Description seems to combine 2 similar names, maybe iProduct and iInterface? Not sure.

Reply:
I'm looking on my phone since I'm about to take a flight back to the states so I may have missed something, but it looks like pyserial is reading it from sysfs in its serial/tools/list_ports_linux.py

Might have to look at the kernel source to see how it's constructed.

Reply:
Tried to order one of the crownstone usb devices to test with but unfortunately "We don’t offer shipping to Hawaii."